### PR TITLE
Display summary of broken links on edit steps page

### PR DESCRIPTION
**Pull request by jonodrew**
This commit adds a method to the Step model. It retrieves the most recent batch_id associated with the Step. Then it retrieves the full batch report from the link-checker-api service. Any broken links are returned to the frontend as an array of Hashes with the same structure as a [`BatchReport`](https://github.com/alphagov/link-checker-api/blob/master/docs/api.md#batchreport-entity)

We're only saving the batch_id for efficiency. There's a tradeoff between storing the whole batch locally against fetching it over the wire, but I've tried to keep the data in one place rather than replicating it in link-checker-api and in collections-publisher.

[Ticket](https://trello.com/c/yZLz7LTB/765-display-the-broken-link-report-in-the-edit-step-page)

### CODE REVIEW

File: spec/models/step_spec.rb [hunk 1]
```diff
@@ -38,4 +40,52 @@
       expect(step_item.errors).to have_key(:logic)
     end
   end
+  describe 'link reports' do
```
Comment (leenagupte) [inline spec/models/step_spec.rb]: This should be `broken_links`, because that's what is is being tested.

File: spec/models/step_spec.rb [hunk 2]
```diff
@@ -38,4 +40,52 @@
...
+    it 'should return nothing if there are no link reports yet' do
+      expect(step_item.broken_links).to be_nil
+    end
+
+    it 'should contain one item if there are link reports and at least one is broken' do
```
Comment (leenagupte) [inline spec/models/step_spec.rb]: Can you add a test that checks that `nil` is returned if a broken link report has been created, but there aren't any broken links.

You only need to create one link report to test this.

File: spec/models/step_spec.rb [hunk 3]
```diff
@@ -38,4 +40,52 @@
...
+            "suggested_fix": "null"
+          }
+        ]
+        )
+      create(:link_report, batch_id: 1, step_id: step_item.id)
+      create(:link_report, completed: Time.now, batch_id: 2, step_id: step_item.id)
+      broken_link_report = step_item.broken_links.first
+      expect(broken_link_report.fetch('uri')).to eq "https://www.gov.uk/404"
+      expect(broken_link_report.fetch('problem_summary')).to eq "404 error (page not found)"
+      expect(step_item.broken_links.length).to eql 1
```
Comment (leenagupte) [inline spec/models/step_spec.rb]: I think these assertions could be split up amongst multiple tests, so it's clear what piece of test data relates to an expectation.

For example, why does the second link report have a completed time? How is that being used/tested?
I think this test only needs to check that the most recent link report is being queried if more than one link report exists. 

You could check the values of the broken links being returned in another test. For that, you only need to stub one link report.

File: spec/models/step_spec.rb [hunk 4]
```diff
@@ -38,4 +40,59 @@
...
+            "warnings": [],
+            "problem_summary": "null",
+            "suggested_fix": "null"
+          }
+        ]
+      )
+      expect(step_item.broken_links).to eql []
+    end
+
+    it 'should contain one item if there are link reports and at least one is broken' do
```
Comment (leenagupte) [inline spec/models/step_spec.rb]: Is this two tests hiding as one? One test to check that broken links are being returned and one test to check that the most recent link report is being returned?
